src/mlpack/methods/preprocess/image_converter_main.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file image_converter_main.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to load and save a image dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core.hpp>
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+using namespace mlpack::data;
+
+
+PROGRAM_INFO("Image Converter",
+    // Short description.
+    "A utility to load an image or set of images into a single dataset that"
+    "can then be used by other mlpack methods and utilities. This can also"
+    "unpack an image dataset into individual files.",
+    // Long description.
+    "This utility takes a image or an array of images and loads them to a"
+    " matrix. You can optionally specify the height " +
+    PRINT_PARAM_STRING("height") + " width " + PRINT_PARAM_STRING("width")
+    + " and channel " + PRINT_PARAM_STRING("channel") + " of the images that"
+    "needs to be loaded; otherwise, these parameters will be automatically"
+    "detected from the image."
+    "\n"
+    "There are other options too, that can be specified such as " +
+    PRINT_PARAM_STRING("quality")
+    + ".\n\n" +
+    "You can also provide a dataset and save them as images using " +
+    PRINT_PARAM_STRING("dataset") + " and " + PRINT_PARAM_STRING("save") +
+    " as an parameter. An example to load an image : "  +
+    "\n\n" +
+    PRINT_CALL("image_converter", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "output", "Y") +
+    "\n\n" +
+    " An example to save an image is :" +
+    "\n\n" +
+    PRINT_CALL("image_converter", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "dataset", "Y", "save", true),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// DEFINE PARAM
+PARAM_VECTOR_IN_REQ(string, "input", "Image filenames which have to "
+    "be loaded/saved.", "i");
+
+PARAM_INT_IN("width", "Width of the image.", "w", 256);
```

Comment:
```suggestion
PARAM_INT_IN("width", "Width of the image.", "w", 0);
```
I think setting the default to 0 here is better, so that the user understands that they have to specify the parameter.

Reply:
Yes, done